src/extension.ts

```diff
@@ -151,12 +151,12 @@ async function startLanguageServer() {
     const serverArgsDebug = ['--startup-file=no', '--history-file=no', '--depwarn=no', `--project=${envForLSPath}`, 'main.jl', jlEnvPath, '--debug=yes', telemetry.getCrashReportingPipename(), oldDepotPath, g_context.globalStoragePath]
     const spawnOptions = {
         cwd: path.join(g_context.extensionPath, 'scripts', 'languageserver'),
-        env: {
-            JULIA_DEPOT_PATH: path.join(g_context.extensionPath, 'scripts', 'languageserver', 'julia_pkgdir'),
+        env: Object.assign(process.env, {
+            JULIA_DEPOT_PATH: path.join(g_context.extensionPath, 'scripts', 'languageserver', 'julia_pkgdir') + ':' + oldDepotPath,

```

Comment:
Are you sure appending `oldDepotPath` is required for this to fix your problem? This process shouldn't ever load random packages, and Base Julia shouldn't care about the DEPOT_PATH at all, no?

Reply:
In my understanding the launched `julia` process needs to be identical to any other `julia` REPL launched on a given system to able to interact with other system libraries. My experience with not including `JULIA_DEPOT` path was all packages got re-installed into the directory where `code-server` had been running from -- re-introducing the original JULIA_DEPOT and a `:` colon reverted back to the original, expected behaviour of julia REPL; now launched from `code-server`

Is including `oldDepotPath` against something I am not aware of?

Reply:
This process isn't user visible and should only load packages present in `scripts/packages`, so I don't think the behaviour you were seeing really was because of this.

Reply:
OK, let me know when you identified the reason.

Reply:
Ok, then let's just remove this for now so the fix for the MKL issue can be merged. We haven't gotten any reports about `DEPOT_PATH` issues with the LS process and I'm not seeing any myself.